src/app/core/campaign/campaign-store.service.ts

```diff
@@ -139,34 +150,47 @@ export class CampaignStoreService {
       const startDate = new Date(flight.startAt.valueOf()).toISOString().slice(0, 10);
       const endDate = new Date(flight.endAt.valueOf()).toISOString().slice(0, 10);
       const inventoryId = flight.set_inventory_uri.split('/').pop();
-      const loading = forkJoin(flight.zones.map((zoneName) => {
-        return this.inventoryService.getInventoryAvailability({
-          id: inventoryId,
-          startDate,
-          endDate,
-          zoneName,
-          flightId: flight.id
+      const loading = forkJoin(
+        flight.zones.map(zoneName => {
+          return this.inventoryService.getInventoryAvailability({
+            id: inventoryId,
+            startDate,
+            endDate,
+            zoneName,
+            flightId: flight.id
+          });
+        })
+      ).pipe(share());
+      loading
+        .pipe(
+          first(),
+          withLatestFrom(this._campaign$)
+        )
+        .subscribe(([availabilities, state]) => {
+          const updatedState = {
+            ...state,
+            availability: {
+              ...state.availability,
+              ...availabilities.reduce((acc, availability) => ({ ...acc, [`${flight.id}-${availability.zone}`]: availability }), {})
+            }
+          };
+          this._campaign$.next(updatedState);
         });
-      })).pipe(
-        share()
-      );
-      loading.pipe(first(), withLatestFrom(this._campaign$)).subscribe(([availabilities, state]) => {
-        const updatedState = {
-          ...state,
-          availability: {
-            ...state.availability,
-            ...availabilities.reduce((acc, availability) => ({...acc, [`${flight.id}-${availability.zone}`]: availability}), {})
-          }
-        };
-        this._campaign$.next(updatedState);
-      });
       return loading;
     }
   }
 
   storeCampaign(): Observable<CampaignStateChanges> {
     const saving = this.putCampaign().pipe(
       switchMap(campaignChanges => {
+        const { flights } = campaignChanges.state;
+        const deletedFlightIds = Object.keys(flights).filter(fId => flights[fId].softDeleted);
+        deletedFlightIds.forEach(fId => {
+          if (flights[fId].remoteFlight) {
+            this.campaignService.deleteFlight(fId).subscribe();
```

Comment:
code smell to subscribe inside an observable stream, typically you would return the observable and use the result in the final subscribe

Reply:
Good shout, meant to circle back to this.